algo/src/main/java/com/hartwig/serve/sources/ckb/ActionableEvidenceFactory.java

```diff
@@ -259,6 +258,28 @@ static EvidenceDirection resolveDirection(@Nullable String direction) {
         return null;
     }
 
+    @NotNull
+    @VisibleForTesting
+    static String extractEvidenceYear(@NotNull LocalDate entryDate, @NotNull List<Reference> references, Therapy therapy) {
```

Comment:
I am quite sure you can make this into a @Nullable Integer!

Also, therapy should be annotated as `Nullable`

Reply:
I think the return value of the function should remain NotNull right? Since you fall back on entryDate if there is no referenceDate and no therapy(Date)?